File: K47 User Manual/1_sound_and_light_displays/07_laser/Code/Python/laser.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import time

LaserPin = 11    # pin11
LedPin = 16

def setup():
	GPIO.setmode(GPIO.BOARD)       # Numbers GPIOs by physical location
	GPIO.setup(LaserPin, GPIO.OUT)   # Set LaserPin's mode is output
	GPIO.setup(LedPin, GPIO.OUT)   # Set LedPin's mode is output
	

def loop():
	while True:
		print '...LaserPin on'
		GPIO.output(LaserPin, GPIO.HIGH)  # LaserPin on
		GPIO.output(LedPin, GPIO.HIGH)
		time.sleep(0.5)
		print 'LaserPin off...'
		GPIO.output(LaserPin, GPIO.LOW) # LaserPin off
		GPIO.output(LedPin, GPIO.LOW)
		time.sleep(0.5)

def destroy():
	GPIO.output(LaserPin, GPIO.LOW)     # LaserPin off
	GPIO.cleanup()                     # Release resource

if __name__ == '__main__':     # Program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()

File: K47 User Manual/1_sound_and_light_displays/07_laser/Code/C/laser.c
#include <wiringPi.h>
#include <stdio.h>

#define LaserPin    0
#define	LedPin		4

int main(void)
{
	if(wiringPiSetup() == -1)
	{
		printf("setup wiringPi failed !");
		return -1; 
	}

	pinMode(LaserPin, OUTPUT);
	pinMode(LedPin, OUTPUT);
	while(1)
	{
		digitalWrite(LaserPin, HIGH);
		digitalWrite(LedPin, HIGH);
		printf("Laser on....\n");
		delay(1000);
		
		digitalWrite(LaserPin, LOW);
		digitalWrite(LedPin, LOW);
		printf("Laser off....\n");
		delay(1000);
	}

	return 0;
}
